k2/csrc/ragged_ops.cu

```diff
@@ -1871,4 +1871,144 @@ Array1<int32_t> CoveringShapeForwardMap(RaggedShape &src,
   return ans;
 }
 
+
+
+void RaggedShapeAxis0Splitter::Init(RaggedShape &src) {
+  int32_t num_layers = src.NumLayers(),
+    num_layers_out = num_layers - 1,
+              dim0 = src.Dim0();
+  K2_CHECK_LE(num_layers_out, 4);  // If this fails, add something to the 4s and
```

Comment:
`4` and `5` are magic numbers here. Shall we given them a name, e.g.,  place the following code
```cpp
constexpr int32_t kMaxNumLayersOut = 4;
```
in the header?

Reply:
OK with me, someone could do that, I've had compilation issues with constexpr in the past so am cautious about it.